apps/remote_control/test/lexical/remote_control/build/error_test.exs

```diff
@@ -452,14 +383,24 @@ defmodule Lexical.RemoteControl.Build.ErrorTest do
         end
         /
         |> compile()
-        |> diagnostic()
+        |> diagnostics()
 
-      assert diagnostic.message =~ "unknown key :c for struct Foo"
+      if Features.span_in_diagnostic?() do
+        [undefined, unknown] = diagnostics
+        assert unknown.message == "unknown key :c for struct Foo"
+        assert unknown.position == {7, 19}
 
-      if Features.with_diagnostics?() do
-        assert diagnostic.position == {7, 19}
+        assert undefined.message == "variable \"c\" is unused"
+        assert undefined.position == {7, 27, 7, 28}
```

Comment:
Don't do assertions like this. Instead assert by extracting ranges and comparing against strings.